tests/p2p/test_peer.py

```diff
@@ -47,6 +47,9 @@ def test_sub_protocol_selection():
     assert isinstance(proto, LESProtocolV2)
     assert proto.cmd_id_offset == peer.base_protocol.cmd_length
 
+    proto = peer.select_sub_protocol([('unknown', 1)])
```

Comment:
This looks like another API that I would be more comfortable with if it raised an exception to preven *accidental* mishandling of the return value in cases where it returns `None`.